test/compilationTests/gnosis/Utils/Math.sol

```diff
@@ -94,12 +94,18 @@ library Math {
         zpow = zpow * z / ONE;
         result += 0x9c7 * zpow / ONE;
         if (shift >= 0) {
-            if (result >> (256-shift) > 0)
+            uint tempres;
+            if (shift > 256)
+                tempres = result << uint(shift - 256);
```

Comment:
Does this contract rely on this behaviour?

Reply:
@hrkrshnn knows this better probably, as he suggested to change this.
This behavior was added because of values of shift that are greater than 256.

Reply:
@chriseth The contract did in fact rely on right-shifting a negative number becoming a left shift. It's a bit sad this has to be rewritten, because I found that code to be quite clever.

Reply:
But it does not work - does it? We have a runtime check for negative shift amounts.

Reply:
I checked, it doesn't work with negative values, so @hrkrshnn, it fails in runtime in case of negative values of shift.
If contract was relying on that, it was wrong than. 

Fixing this now

Reply:
@mijovic I looked at this contract again. The value of `shift` would be bounded between `-64` and `192`. So the `uint()` cast would actually work. Sorry about that.